ci: Occasional errors occur when downloading models in CI.

Such as: https://github.com/vllm-project/semantic-router/actions/runs/17833363856/job/50704034026?pr=171

This has occurred frequently in the recent period.

logs:

```
huggingface_hub.errors.LocalEntryNotFoundError: An error happened while trying to locate the files on the Hub and we cannot find the appropriate snapshot folder for the specified revision on the local disk. Please check your internet connection and try again.
make[1]: *** [tools/make/models.mk:20: download-models] Error 1
make[1]: Leaving directory '/home/runner/work/semantic-router/semantic-router'
make: *** [Makefile:5: _run] Error 2
```

Checking the log is due to network issues.

--- a/.github/workflows/test-and-build.yml
+++ b/.github/workflows/test-and-build.yml
@@ -51,6 +51,16 @@ jobs:
           ~/go/pkg/mod
         key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
 
+    - name: Cache Models
+      uses: actions/cache@v4
+      with:
+        path: |
+          models/
+          ~/.cache/huggingface/
+        key: ${{ runner.os }}-models-v1-${{ hashFiles('tools/make/models.mk') }}
+        restore-keys: |
+          ${{ runner.os }}-models-v1-
+
     - name: Check go mod tidy
       run: make check-go-mod-tidy
 

